### PR TITLE
Task/setup typescript

### PR DESCRIPTION
**Pull request by Wiecek-K**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "target": "es6",
+    "noImplicitAny": true,
+    "moduleResolution": "node",
+    "sourceMap": true,
+    "outDir": "",
```
Comment (Sumick) [inline tsconfig.json]: **nitpick:** Create `dist` as out dir, or any other name - just to not mix it together with source files

File: package.json [hunk 2]
```diff
@@ -10,8 +10,25 @@
     "server": "node index.js"
```
Comment (Sumick) [inline package.json]: **nitpick:** You are missing some `dev` script, so it compiles your TS files into JS on the go. Check out this: https://github.com/wclr/ts-node-dev

**Review by Sumick — CHANGES_REQUESTED** [review]
You are missing 2 important things, I've listed them below

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### .eslintrc.js
- The addition of `@typescript-eslint/parser` and `@typescript-eslint/recommended` in the `extends` array indicates that TypeScript linting rules are being used, which aligns with the task goal of setting up TypeScript in the project.
- The `parserOptions` object specifies the ECMAScript version and source type, which is appropriate.
- The rules seem to be configured correctly for TypeScript, with some exceptions and warnings for specific rules.
- No bugs or missed best practices found.

### .prettierrc.json
- The added configuration seems to be fine, specifying the desired formatting rules.
- No bugs or missed best practices found.

### index.js
- The modification changes the single quotes to double quotes in the `console.log` statement.
- No bugs or missed best practices found.

### package.json
- The addition of dev dependencies for TypeScript and ESLint with TypeScript support indicates that TypeScript is being set up in the project.
- The new scripts `dev` and `lint` are added, which align with the task requirements of being able to write `.ts` files and refactoring existing `.js` files to `.ts`.
- No bugs or missed best practices found.

### src/app.ts
- The file has been renamed from `.js` to `.ts`, which aligns with the task requirement of refactoring existing `.js` files to `.ts`.
- The imports have been updated to use TypeScript syntax.
- The `dotenv` import has been changed to use the default import syntax.
- The `mongodbKey` and `mongodbUser` variables have been removed and replaced with `process.env.MONGODB_URI`, which aligns with the task requirement of using environment variables.
- The `express` import has been changed to use TypeScript syntax.
- The request and response types have been explicitly specified in the `app.get` route handler, which is a good practice in TypeScript.
- No bugs or missed best practices found.

### tsconfig.json
- The added `tsconfig.json` file indicates that TypeScript is being set up in the project.
- The compiler options seem to be configured correctly, specifying the module system, target version, and other necessary settings.
- The `include` property specifies the source files to be included in the compilation.
- No bugs or missed best practices found.

Overall, the code changes align with the task goal of setting up TypeScript in the project. The necessary configuration files have been added or modified, and the existing code has been refactored to use TypeScript syntax. No bugs or missed best practices were found in the code changes.

**Pull request merged**